sdk/src/wallet/wallet/builder.rs

```diff
@@ -191,7 +194,11 @@ impl WalletBuilder {
         let event_emitter = Arc::new(Mutex::new(EventEmitter::new()));
 
         #[cfg(feature = "storage")]
-        let accounts = storage_manager.lock().await.get_accounts().await.unwrap_or_default();
+        let mut accounts = storage_manager.lock().await.get_accounts().await.unwrap_or_default();
+        // It happened that inputs got locked, the transaction failed, but they weren't unlocked again, so we do this
+        // here
+        #[cfg(feature = "storage")]
+        unlock_unused_inputs(&mut accounts)?;
```

Comment:
Is it enough to do that only when building a wallet ?

Reply:
I think so, normally this shouldn't even happen, that's why I also added more matches on the errors to unlock the inputs if something in the transaction process fails
Also tried to clean it up during syncing, but that doesn't work in a safe way with the way we create the transactions. To make that safe we would need to add a new lock or change it so the account gets locked the whole time